lib/stat_tracker.rb

```diff
@@ -11,4 +11,53 @@ def initialize(data)
     @data = data
   end
 
+  def games_coached(season_id)
+    coaches = []
+    season = CSV.read(@data[:games], headers: true, header_converters: :symbol).find_all {|row| row[:season] == season_id}
+    games = CSV.read(@data[:game_teams], headers: true, header_converters: :symbol)
+    season.each do |season|
+      games.each do |game|
+        if season[:game_id] == game[:game_id]
+          coaches << game[:head_coach]
```

Comment:
This code works, but is what I referenced as being expensive. It's essentially taking one line of the season CSV file, scanning every single line of the games file, then moving on to the second line of the season CSV file and scanning every single line, and so on. And considering there are like 15,000 rows of data, this is definitely not how to do it. Working on solution now.

Reply:
Yeah so this is what I mentioned earlier in Slack (and no worries if you missed it). If you put your CSV.read methods inside initialize, it only has to do it once. That's what I learned from your mentor, so I tried that on mine and it works.

Reply:
I have it initializing in self with .from_csv. He told me that was the way to go, but I'll see if putting it in with initialize speeds up the process further!